splunk_connect_for_snmp/inventory/loader.py

```diff
@@ -60,9 +62,98 @@
 MONGO_DB = os.getenv("MONGO_DB", "sc4snmp")
 CONFIG_PATH = os.getenv("CONFIG_PATH", "/app/config/config.yaml")
 INVENTORY_PATH = os.getenv("INVENTORY_PATH", "/app/inventory/inventory.csv")
+CONFIG_FROM_MONGO = os.getenv("CONFIG_FROM_MONGO", "false")
+INVENTORY_KEYS_TRANSFORM = {
+    "securityEngine": "security_engine",
+    "SmartProfiles": "smart_profiles",
+}
+BOOLEAN_INVENTORY_FIELDS = ["delete", "smart_profiles"]
 CHAIN_OF_TASKS_EXPIRY_TIME = int(os.getenv("CHAIN_OF_TASKS_EXPIRY_TIME", "60"))
 
 
+def configure_ui_database(mongo_client):
+    """
+    If the UI wasn't used during previous update, and now it is used, create UI collections in Mongo
+    with config from files. Similarly, if the UI was used previous update, and now it isn't used,
+    drop UI collections in Mongo.
+    """
+    used_ui_collection = mongo_client.sc4snmp.used_ui
+    used_ui_doc = used_ui_collection.find_one()
+    if used_ui_doc:
+        used_ui = used_ui_doc["used_ui"]
+    else:
+        used_ui = False
+
+    if CONFIG_FROM_MONGO.lower() in ["true", "1", "t"] and not used_ui:
+        inventory_ui_collection = mongo_client.sc4snmp.inventory_ui
+        groups_ui_collection = mongo_client.sc4snmp.groups_ui
+        profiles_ui_collection = mongo_client.sc4snmp.profiles_ui
+        used_ui_collection.update_one({}, {"$set": {"used_ui": True}}, upsert=True)
+
+        with open(INVENTORY_PATH, encoding="utf-8") as csv_file:
+            ir_reader = DictReader(csv_file)
+            for inventory_line in ir_reader:
+                for key in INVENTORY_KEYS_TRANSFORM.keys():
+                    if key in inventory_line:
+                        new_key = INVENTORY_KEYS_TRANSFORM[key]
+                        inventory_line[new_key] = inventory_line.pop(key)
+
+                for field in BOOLEAN_INVENTORY_FIELDS:
+                    if inventory_line[field].lower() in ["", "f", "false", "0"]:
+                        inventory_line[field] = False
+                    else:
+                        inventory_line[field] = True
+
+                port = (
+                    int(inventory_line["port"])
+                    if len(inventory_line["port"]) > 0
+                    else 161
```

Comment:
changed

Reply:
I see it's still `if inventory_line["port"] > 0` and it can be just `if inventory_line["port"]` as according to previous logic we only checked if length is greater than 0, so if it's not empty, not about what value it contains. I don't thing someone will pass negative value as port, so we can check just if it's not empty.

Reply:
Right, I forgot to delete a comparison. Now it should be done